db/silkworm/etl/collector.hpp

```diff
@@ -36,8 +36,10 @@ class Collector {
     Collector(const Collector&) = delete;
     Collector& operator=(const Collector&) = delete;
 
-    explicit Collector(const char* work_path = nullptr, size_t optimal_size = kOptimalBufferSize)
-        : work_path_{set_work_path(work_path)}, buffer_{optimal_size} {}
+    explicit Collector(const std::filesystem::path work_path, size_t optimal_size = kOptimalBufferSize)
+        : work_path_managed_{true}, work_path_{set_work_path(work_path)}, buffer_{optimal_size} {}
```

Comment:
Shouldn't `work_path_managed_` be the other way round? I'd think that it should be true when not specified externally and created internally, and false when specified externally.

Reply:
Damn you're right. I need a couple of days off